core/genesis.go

```diff
@@ -596,28 +596,40 @@ func DefaultHoleskyGenesisBlock() *Genesis {
 }
 
 // DeveloperGenesisBlock returns the 'geth --dev' genesis block.
-func DeveloperGenesisBlock(gasLimit uint64, faucet common.Address) *Genesis {
+func DeveloperGenesisBlock(gasLimit uint64, faucet common.Address, genesisAlloc *GenesisAlloc) *Genesis {
 	// Override the default period to the user requested one
 	config := *params.AllDevChainProtocolChanges
+	alloc := map[common.Address]GenesisAccount{
+		common.BytesToAddress([]byte{1}): {Balance: big.NewInt(1)}, // ECRecover
+		common.BytesToAddress([]byte{2}): {Balance: big.NewInt(1)}, // SHA256
+		common.BytesToAddress([]byte{3}): {Balance: big.NewInt(1)}, // RIPEMD
+		common.BytesToAddress([]byte{4}): {Balance: big.NewInt(1)}, // Identity
+		common.BytesToAddress([]byte{5}): {Balance: big.NewInt(1)}, // ModExp
+		common.BytesToAddress([]byte{6}): {Balance: big.NewInt(1)}, // ECAdd
+		common.BytesToAddress([]byte{7}): {Balance: big.NewInt(1)}, // ECScalarMul
+		common.BytesToAddress([]byte{8}): {Balance: big.NewInt(1)}, // ECPairing
+		common.BytesToAddress([]byte{9}): {Balance: big.NewInt(1)}, // BLAKE2b
+	}
+
+	if genesisAlloc != nil {
+		for addr, account := range *genesisAlloc {
+			// Don't allow overwriting the predefined accounts (precompiles)
+			if _, ok := alloc[addr]; ok {
+				log.Crit("Overwriting precompiles disallowed", "address", addr)
+			}
+			alloc[addr] = account
+		}
+	}
+
+	alloc[faucet] = GenesisAccount{Balance: new(big.Int).Sub(new(big.Int).Lsh(big.NewInt(1), 256), big.NewInt(9))}
```

Comment:
We should accumulate the balances dished out to the user input and subtract that from the faucet. Otherwise we can end up with an account overflowing 2^256 and that's gonna end up very weird.

Reply:
I see, fix it ASAP.